public/app/features/dashboard-scene/scene/PanelTimeRange.tsx

```diff
@@ -0,0 +1,145 @@
+  hideTimeOverride?: boolean;
+  timeInfo?: string;
+}
+
+export class PanelTimeRange extends SceneObjectBase<PanelTimeRangeState> implements SceneTimeRangeLike {
+  public static Component = PanelTimeRangeRenderer;
+
+  public constructor(state: Partial<PanelTimeRangeState> = {}) {
+    super({
+      ...state,
+      // This time range is not valid until activation
+      from: 'now-6h',
+      to: 'now',
+      value: getDefaultTimeRange(),
+    });
+
+    this.addActivationHandler(() => this._activationHandler());
+  }
+
+  private getTimeOverride(parentTimeRange: TimeRange): TimeOverrideResult {
+    const { timeFrom, timeShift } = this.state;
+    const newTimeData = { timeInfo: '', timeRange: parentTimeRange };
+
+    if (timeFrom) {
+      const timeFromInterpolated = sceneGraph.interpolate(this, this.state.timeFrom);
+      const timeFromInfo = rangeUtil.describeTextRange(timeFromInterpolated);
+
+      if (timeFromInfo.invalid) {
+        newTimeData.timeInfo = 'invalid time override';
+        return newTimeData;
+      }
+
+      // Only evaluate if the timeFrom if parent time is relative
+      if (rangeUtil.isRelativeTimeRange(parentTimeRange.raw)) {
+        newTimeData.timeInfo = timeFromInfo.display;
+        newTimeData.timeRange = {
+          from: dateMath.parse(timeFromInfo.from)!,
+          to: dateMath.parse(timeFromInfo.to)!,
+          raw: { from: timeFromInfo.from, to: timeFromInfo.to },
+        };
+      }
+    }
+
+    if (timeShift) {
+      const timeShiftInterpolated = sceneGraph.interpolate(this, this.state.timeShift);
+      const timeShiftInfo = rangeUtil.describeTextRange(timeShiftInterpolated);
+
+      if (timeShiftInfo.invalid) {
+        newTimeData.timeInfo = 'invalid timeshift';
+        return newTimeData;
+      }
+
+      const timeShift = '-' + timeShiftInterpolated;
+      newTimeData.timeInfo += ' timeshift ' + timeShift;
+      const from = dateMath.parseDateMath(timeShift, newTimeData.timeRange.from, false)!;
+      const to = dateMath.parseDateMath(timeShift, newTimeData.timeRange.to, true)!;
+
+      newTimeData.timeRange = { from, to, raw: { from, to } };
+    }
+
+    return newTimeData;
+  }
+
+  private _activationHandler(): void {
+    const parentTimeRange = this.getParentTimeRange();
+
+    this._subs.add(parentTimeRange.subscribeToState((state) => this.handleParentTimeRangeChanged(state.value)));
+
+    this.handleParentTimeRangeChanged(parentTimeRange.state.value);
+  }
+
+  private handleParentTimeRangeChanged(parentTimeRange: TimeRange) {
+    const overrideResult = this.getTimeOverride(parentTimeRange);
+    this.setState({ value: overrideResult.timeRange, timeInfo: overrideResult.timeInfo });
+  }
+
+  private getParentTimeRange(): SceneTimeRangeLike {
+    if (!this.parent || !this.parent.parent) {
+      throw new Error('Missing parent');
+    }
+
+    // Need to go up two levels otherwise we will get ourselves
+    return sceneGraph.getTimeRange(this.parent.parent);
+  }
```

Comment:
Same as https://github.com/grafana/scenes/blob/main/packages/scenes/src/core/SceneTimeZoneOverride.tsx#L26

Time to have this as a core class that is extended? Especially that this re-implements onTimeRangeChange/onRefresh/onTimeZoneChange/getTimeZone.

Reply:
But also we have two cases for this now, not sure a new one will emerge.